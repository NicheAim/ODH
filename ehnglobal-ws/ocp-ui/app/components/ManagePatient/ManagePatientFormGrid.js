import { Grid } from 'styled-css-grid';

const ManagePatientFormGrid = Grid.extend`
  padding-left: 0.5vw;
  margin: 2vh 1vw;
  grid-column-gap: 2vw;
  grid-row-gap: 2vh;
  grid-template-columns: 1fr;
  grid-template-areas:
    'generalInformationSubtitle'
    'contextGroup'
    'eligible'
    'firstName'
    'lastName'
    'birthDate'
    'genderCode'
    'birthSex'
    'race'
    'ethnicity'
    'language'
    'identifierGroup'
    'medicalId'
    'addresses'
    'contacts'
    'emergency-contact'
    'flags'
    'episodeOfCares'
    'coverages'
    'buttonGroup';

  @media only screen and (min-width: 768px) {
    grid-template-columns: 1fr 1fr;
    grid-template-areas:
      'generalInformationSubtitle generalInformationSubtitle'
      'contextGroup contextGroup'
      'eligible eligible'
      'firstName lastName'
      'birthDate genderCode'
      'birthSex race'
      'ethnicity language'
      'identifierGroup identifierGroup'
      'medicalId medicalId'
      'contacts contacts'
      'emergency-contact emergency-contact'
      'addresses addresses'
      'flags flags'
      'episodeOfCares episodeOfCares'
      'coverages coverages'
      'buttonGroup .';
  }

  @media only screen and (min-width: 1200px) {
    grid-template-columns: repeat(12, 1fr);
    grid-template-areas:
      'generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle'
      'contextGroup contextGroup contextGroup contextGroup contextGroup contextGroup contextGroup contextGroup contextGroup contextGroup contextGroup contextGroup'
      'eligible . . . . . . . . . . .'
      'firstName firstName firstName firstName lastName lastName lastName lastName . . . .'
      'birthDate birthDate birthDate birthDate genderCode genderCode birthSex birthSex . . . .'
      'race race ethnicity ethnicity ethnicity language language . . . . .'
      'identifierGroup identifierGroup identifierGroup identifierGroup identifierGroup . . . . . . .'
      'medicalId medicalId medicalId medicalId medicalId . . . . . . .'
      'contacts contacts contacts contacts contacts contacts contacts contacts contacts contacts contacts contacts'
      'emergency-contact emergency-contact emergency-contact emergency-contact emergency-contact emergency-contact emergency-contact emergency-contact emergency-contact emergency-contact emergency-contact emergency-contact'
      'addresses addresses addresses addresses addresses addresses addresses addresses addresses addresses addresses addresses'
      'flags flags flags flags flags flags flags flags flags flags flags flags'
      'episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares'
      'coverages coverages coverages coverages coverages coverages coverages coverages coverages coverages coverages coverages'
      'buttonGroup buttonGroup buttonGroup buttonGroup . . . . . . . .';
  }
`;

ManagePatientFormGrid.propTypes = Grid.propTypes;

export default ManagePatientFormGrid;
