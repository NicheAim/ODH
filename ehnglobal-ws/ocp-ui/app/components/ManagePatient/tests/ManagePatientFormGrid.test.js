import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'jest-styled-components';

import ManagePatientFormGrid from '../ManagePatientFormGrid';

configure({ adapter: new Adapter() });

describe('<ManagePatientFormGrid />', () => {
  describe('snapshot tests', () => {
    it('should match snapshot', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<ManagePatientFormGrid>{children}</ManagePatientFormGrid>);

      // Assert
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe('structural tests', () => {
    it('should have children', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<ManagePatientFormGrid>{children}</ManagePatientFormGrid>);

      // Assert
      expect(renderedComponent.contains(children)).toEqual(true);
    });
  });

  describe('style tests', () => {
    it('should be grid', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<ManagePatientFormGrid>{children}</ManagePatientFormGrid>);

      // Assert
      expect(renderedComponent).toHaveStyleRule('display', 'grid');
    });

    it('should have default styles', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<ManagePatientFormGrid>{children}</ManagePatientFormGrid>);

      // Assert
      expect(renderedComponent).toHaveStyleRule('padding-left', '0.5vw');
      expect(renderedComponent).toHaveStyleRule('margin', '2vh 1vw');
      expect(renderedComponent).toHaveStyleRule('grid-column-gap', '2vw');
      expect(renderedComponent).toHaveStyleRule('grid-row-gap', '2vh');
      expect(renderedComponent).toHaveStyleRule('grid-template-columns', '1fr');
      expect(renderedComponent).toHaveStyleRule('grid-template-areas', '"generalInformationSubtitle"    "contextGroup"    "firstName"    "lastName"    "birthDate"    "genderCode"    "birthSex"    "race"    "ethnicity"    "language"    "identifierGroup"    "addresses"    "contacts"    "flags"    "episodeOfCares"    "coverages"    "buttonGroup"');
    });

    it('should have styles in min-width: 768px', () => {
      // Arrange
      const children = <div>test</div>;
      const media = 'only screen and (min-width: 768px)';

      // Act
      const renderedComponent = shallow(<ManagePatientFormGrid>{children}</ManagePatientFormGrid>);

      // Assert
      expect(renderedComponent).toHaveStyleRule('grid-template-columns', '1fr 1fr', { media });
      expect(renderedComponent).toHaveStyleRule('grid-template-areas', '"generalInformationSubtitle generalInformationSubtitle"      "contextGroup contextGroup"      "firstName lastName"      "birthDate genderCode"      "birthSex race"      "ethnicity language"      "identifierGroup identifierGroup"      "contacts contacts"      "addresses addresses"      "flags flags"      "episodeOfCares episodeOfCares"      "coverages coverages"      "buttonGroup ."', { media });
    });

    it('should have styles in min-width: 1200px', () => {
      // Arrange
      const children = <div>test</div>;
      const media = 'only screen and (min-width: 1200px)';

      // Act
      const renderedComponent = shallow(<ManagePatientFormGrid>{children}</ManagePatientFormGrid>);

      // Assert
      expect(renderedComponent).toHaveStyleRule('grid-template-columns', 'repeat(12, 1fr)', { media });
      expect(renderedComponent).toHaveStyleRule('grid-template-areas', '"generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle generalInformationSubtitle"      "contextGroup contextGroup contextGroup contextGroup contextGroup contextGroup contextGroup contextGroup contextGroup contextGroup contextGroup contextGroup"      "firstName firstName firstName firstName lastName lastName lastName lastName . . . ."      "birthDate birthDate birthDate birthDate genderCode genderCode birthSex birthSex . . . ."      "race race ethnicity ethnicity ethnicity language language . . . . ."      "identifierGroup identifierGroup identifierGroup identifierGroup identifierGroup . . . . . . ."      "contacts contacts contacts contacts contacts contacts contacts contacts contacts contacts contacts contacts"      "addresses addresses addresses addresses addresses addresses addresses addresses addresses addresses addresses addresses"      "flags flags flags flags flags flags flags flags flags flags flags flags"      "episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares episodeOfCares"      "coverages coverages coverages coverages coverages coverages coverages coverages coverages coverages coverages coverages"      "buttonGroup buttonGroup buttonGroup buttonGroup . . . . . . . ."', { media });
    });
  });
});
