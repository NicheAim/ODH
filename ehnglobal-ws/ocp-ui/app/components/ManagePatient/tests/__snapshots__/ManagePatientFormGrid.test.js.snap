// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManagePatientFormGrid /> snapshot tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div>
    <div>
      test
    </div>
  </styled.div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        test
      </div>,
      "className": "sc-bdVaJa bQgoME",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
      },
      "ref": null,
      "rendered": "test",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          test
        </div>,
        "className": "sc-bdVaJa bQgoME",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
